src/core/lib/surface/server.cc

```diff
@@ -943,9 +943,19 @@ static void destroy_channel_elem(grpc_channel_element* elem) {
   channel_data* chand = static_cast<channel_data*>(elem->channel_data);
   if (chand->registered_methods) {
     for (i = 0; i < chand->registered_method_slots; i++) {
-      grpc_slice_unref_internal(chand->registered_methods[i].method);
+      // TODO(arjunroy) : Unref is a nop for ExternallyManagedSlice.
```

Comment:
Added Ref/Unref for this concrete type. Other slices will get the same method eventually.

Reply:
It's not clear to me how this is going to work once we add these methods to all slice types.  What I'd ideally like to see here is the ability to use `RefCountedPtr<>` for slices, so that the compiler can enforce ownership semantics.  But maybe that requires making them virtual methods, which will hurt performance?

In any case, I'd really prefer to leave the `grpc_slice_unref_internal()` calls in place until we have a full solution for all slice types.  I don't care whether that happens as part of this PR or whether we defer that work to a subsequent PR, but I'd rather not merge a change that does one part without the other.

Reply:
Reverted this to grpc_slice_unref_internal().